### PR TITLE
OCPBUGS-14064: Split baremetal-operator into a separate pod

### PR DESCRIPTION
**Pull request by dtantsur**
There is no reason to keep it tied to Ironic. Actually, it causes
constant chicken-and-egg problems where BMO needs to know the location
of Ironic.

### CODE REVIEW

**Comment by dtantsur** [comment]
/retest

I don't understand why github does not register the update here...

File: controllers/provisioning_controller.go [hunk 1]
```diff
@@ -328,7 +329,7 @@ func (r *ProvisioningReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		}
 	}
 
-	// Determine the status of the deployment
+	// Determine the status of the baremetal deployment
```
Comment (zaneb) [inline controllers/provisioning_controller.go:332]: metal3?

**Pull request merged**